go/chat/boxer.go

```diff
@@ -687,6 +690,23 @@ func (b *Boxer) makeHeaderHash(ctx context.Context, headerSealed chat1.SignEncry
 	return b.hashV1(buf.Bytes()), nil
 }
 
+func (b *Boxer) makeBodyHash(bodyCiphertext chat1.EncryptedData) (chat1.Hash, UnboxingError) {
+	buf := bytes.Buffer{}
+	err := binary.Write(&buf, binary.BigEndian, int32(bodyCiphertext.V))
+	if err != nil {
+		return nil, NewPermanentUnboxingError(err)
+	}
+	_, err = buf.Write(bodyCiphertext.N)
```

Comment:
I wonder if we should add an assertion in here that the nonce length has not changed? I know we're assuming it's a constant. (This would apply equally to the other one.)